resources/providers/awslib/multi_region.go

```diff
@@ -100,11 +100,16 @@ func MultiRegionFetch[T any, K any](ctx context.Context, set map[string]T, fetch
 				return
 			}
 			crossRegionResources = append(crossRegionResources, results)
-		}(client, region)
+		}(client, region, errChan)
 	}
 
 	wg.Wait()
-	return crossRegionResources, err
+	select {
+	case err := <-errChan:
+		return crossRegionResources, err
```

Comment:
Another option here is to return a multi-error (From Go 1.20)
https://lukas.zapletalovi.com/posts/2022/wrapping-multiple-errors/

Reply:
Cool! will be useful once we'll run 1.20